doc/install/install-ceph-gateway.rst

```diff
@@ -1,255 +1,583 @@
+directory, you will want to maintain those paths in your Ceph configuration
+file if you used something other than default paths.
 
-	sudo apt-get install openssl ssl-cert
+A typical Ceph Object Gateway configuration file for an Apache-based deployment
+looks something similar as the following::
 
-#. Enable the SSL module::
+On Red Hat Enterprise Linux::
 
-	sudo a2enmod ssl
+ [client.radosgw.gateway-node1]
+ host = {hostname}
+ keyring = /etc/ceph/ceph.client.radosgw.keyring
+ rgw socket path = ""
+ log file = /var/log/radosgw/client.radosgw.gateway-node1.log
+ rgw frontends = fastcgi socket\_port=9000 socket\_host=0.0.0.0
+ rgw print continue = false
 
-#. Generate a certificate::
+On Ubuntu::
 
-	sudo mkdir /etc/apache2/ssl
-	sudo openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout /etc/apache2/ssl/apache.key -out /etc/apache2/ssl/apache.crt
+ [client.radosgw.gateway-node]
+ host = {hostname}
+ keyring = /etc/ceph/ceph.client.radosgw.keyring
+ rgw socket path = /var/run/ceph/ceph.radosgw.gateway.fastcgi.sock
+ log file = /var/log/radosgw/client.radosgw.gateway-node1.log
 
-#. Restart Apache::
+To modify it for use with Civetweb, simply remove the Apache-specific settings
+such as ``rgw_socket_path`` and ``rgw_print_continue``. Then, change the
+``rgw_frontends`` setting to reflect Civetweb rather than the Apache FastCGI
+front end and specify the port number you intend to use. For example::
 
-	sudo service apache2 restart
+ [client.radosgw.gateway-node1]
+ host = {hostname}
+ keyring = /etc/ceph/ceph.client.radosgw.keyring
+ log file = /var/log/radosgw/client.radosgw.gateway-node1.log
+ rgw_frontends = civetweb port=80
 
+Finally, restart the Ceph Object Gateway. On Red Hat Enterprise Linux execute::
 
-See the `Ubuntu Server Guide`_ for additional details.
+ sudo systemctl restart ceph-radosgw.service
 
+On Ubuntu execute::
 
-RPM-based distros
------------------
+ sudo service radosgw restart id=rgw.<short-hostname>
+
+If you used a port number that is not open, you will also need to open that
+port on your firewall.
+
+Configure Bucket Sharding
+-------------------------
 
-To enable SSL on RPM-based distros, execute the following steps:
+A Ceph Object Gateway stores bucket index data in the ``index_pool``, which
+defaults to ``.rgw.buckets.index``. Sometimes users like to put many objects
+(hundreds of thousands to millions of objects) in a single bucket. If you do
+not use the gateway administration interface to set quotas for the maximum
+number of objects per bucket, the bucket index can suffer significant
+performance degradation when users place large numbers of objects into a
+bucket.
 
-#. Ensure that you have installed the dependencies::
+In Ceph 0.94, you may shard bucket indices to help prevent performance
+bottlenecks when you allow a high number of objects per bucket. The
+``rgw_override_bucket_index_max_shards`` setting allows you to set a maximum
```

Comment:
Hi, the option is already there: http://docs.ceph.com/docs/master/radosgw/config-ref/ (search for "rgw override bucket index max shards")